test/expect-tests/dune_console/dune_console_tests.ml

```diff
@@ -193,3 +193,29 @@ let%expect_test "Status line overwriting." =
 Here is a status line\r                     \rHere is another status line\r                           \r
   |}]
 ;;
+
+(* set_directory tests *)
+
+let%expect_test "set_directory with output prints entering and leaving" =
+  let module Console = New () in
```

Comment:
Something I just realised is that this is just aliasing a module and not creating a fresh copy like I had thought at the time.

Reply:
I tried making `set_directory` a Fdecl and couldn't share it between tests because it was mutated. We don't have this issue with references however since we can overwrite them.

Reply:
Even if it's a ref, I think you should forbid this from being set more than once. This is the behavior we rely on, so it's wrong to relax this invariant for a test.